libs/designer-ui/src/lib/utils/utils.ts

```diff
@@ -348,7 +349,7 @@ export const filterRecord = <T>(data: Record<string, T>, filter: (_key: string,
     .reduce((res: any, [key, value]: any) => ({ ...res, [key]: value }), {});
 };
 
-export const getConnectorCategoryString = (connectorId: string): string => {
+export const getConnectorCategoryString = (connector: Connector | OperationApi | string): string => {
```

Comment:
Is there a reason we're keeping this parameter typed as `| string`?  It looks like it should only work on Connector / Api objects now.

And I'm guessing so, but did you adjust all references of this method?

Reply:
There are a handful of places where we still pass a string.

- `libs\designer\src\lib\ui\panel\connectionsPanel\allConnections\connectorConnectionsCard.tsx` &mdash; This can have `connector === undefined` so it falls back on `connectorId`.
- `libs\designer\src\lib\ui\panel\nodeDetailsPanel\tabs\aboutTab.tsx` &mdash; This does not have access to the `Connector` object because it comes from `operationInfo`, which only has the ID.

All other calls have access to the `Connector` or `OperationApi` object.